src/glossary.md

```diff
@@ -0,0 +1,281 @@
+
+## Fuel Core
+
+`fuel-core` is the name of the Fuel client implementation.
+
+## Gas
+
+Gas is a variable fee charged by a node to process a transaction that is executed on-chain.
+
+## Indexer
+
+An indexer is a program that watches and organizes blockchain data so it can be easily queried.
+
+## Input
+
+An input refers to a transaction input, or which state elements (usually: UTXOs) are consumed by a transaction.
+
+## Layer 1 (L1)
+
+Also called a level 1, this refers to a base layer blockchain that is not built on top of any other blockchain.
+
+## Layer 2 (L2)
+
+Also called a level 2, this is a blockchain that is built on top of another blockchain. Layer 2 networks can offer unique benefits like allowing for cheaper transactions or sovereign rollups that can fork without forking the base layer.
+
+## Light Client
+
+A light client is a client that doesn't validate blocks and transactions but still offers some functionality to send transactions.
+
+## Locked Wallet
+
+A locked wallet is a wallet that can only interact with read-only smart contract methods.
+
+## Mainnet
+
+Mainnet refers to the main network of a blockchain, as opposed to a testnet.
+
+## Message
+
+A type of input that only includes specific metadata, and is often used for bridging.
+
+## Mint
+
+Minting refers to the creation of new coins.
+
+## Modular
+
+Referring to a blockchain architecture that allows for execution, settlement, consensus, and data availability to run on separate layers.
+
+## Monolithic
+
+Referring to a blockchain architecture that handles execution, settlement, consensus, and data availability all at the same time on a single layer.
+
+## Native Asset
+
+With Fuel any contract can make its own native asset. On Ethereum, the only native asset is ETH. This allows for much cheaper token transactions because it doesn't require any contract state changes. It also allows you to directly forward any asset in a transaction call, avoiding the need for the `approve` and `transferFrom` mechanisms.
+
+## Network
+
+Another name for a blockchain.
+
+## Node
+
+A client that validates and produces blocks for the network.
+
+## Optimistic Rollup
+
+An optimistic rollup is a sidechain that uses fraud proofs to verify transactions instead of relying on a majority of validators to be honest.
+
+## Output
+
+An output refers to a transaction output, or which state elements are output by a transaction.
+
+## Parallel Transactions
+
+Parallel transactions refers to the ability of the FuelVM to process multiple transactions in parallel.
+
+## Predicate
+
+A predicate is a pure function that can return true or false, and is sent inside a transaction as bytecode and checked at transaction validity time. If it evaluates to `false` the transaction will not be processed, and no gas will be used. If it evaluates to `true`, any coins belonging to the address equal to the root of the predicate bytecode may be spent by the transaction.
```

Comment:
```suggestion
A predicate is a pure function that can return true or false, and is sent inside a transaction as bytecode and checked at transaction validity time. If it evaluates to `false` the transaction will not be processed, and no gas will be used. If it evaluates to `true`, any coins belonging to the address equal to the Merkle root of the predicate bytecode may be spent by the transaction.
```

Perhaps add "Merkle tree" to the glossary? :

"A Merkle tree is a data structure which uses a cryptographic hash function to condense a set of data into a single value, called the root. It allows efficient proofs that a given element is part of the set"

Reply:
Did you see above comment about adding a Merkle tree entry to the glossary? I guess if we're going to introduce the term in explaining another, we should also explain the term itself, just in case.